lighthouse-core/gather/computed/critical-request-chains.js

```diff
@@ -17,11 +17,15 @@ class CriticalRequestChains extends ComputedArtifact {
    * For now, we use network priorities as a proxy for "render-blocking"/critical-ness.
    * It's imperfect, but there is not a higher-fidelity signal available yet.
    * @see https://docs.google.com/document/d/1bCDuq9H1ih9iNjgzyAL0gpwNFiEP4TZS-YLRp_RuMlc
-   * @param  {any} request
+   * @param {any} request
+   * @param {!WebInspector.NetworkRequest}
    */
-  static isCritical(request) {
+  static isCritical(request, mainResource) {
     const resourceTypeCategory = request._resourceType && request._resourceType._category;
 
+    // Iframes are considered High Priority but they are not render blocking
+    const isIframe = resourceTypeCategory === WebInspector.resourceTypes.Document._category
```

Comment:
the other examples in here I believe are a bit more verbose than they need to be, you should be able to just check `request._resourceType === WebInspector.resourceTypes.Document`